library/arith.lisp

```diff
@@ -692,6 +692,17 @@ The fields are defined as follows:
       (lisp String (z)
         (cl:format cl:nil "~D" z))))
 
+  (define-instance (Into Single-Float String)
+    (define (into z)
+      (lisp String (z)
+        (cl:format cl:nil "~D" z))))
```

Comment:
`~D` is for (Decimal) printing of integers.

You might instead opt for `prin1-to-string`, which will try to write a `read`able string

Reply:
ok updated